ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java

```diff
@@ -273,6 +275,10 @@ private ResultAndTerm submitRequestAndWaitResult(Phase phase, RaftConfigurationI
       throws InterruptedException {
     final ResultAndTerm r;
     final Collection<RaftPeer> others = conf.getOtherPeers(server.getId());
+    if (!conf.containsInBothConfs(server.getId())) {
```

Comment:
It should use containsInConf(..) instead of containsInBothConfs(..) since we don't care about the previous conf.